src/main/java/org/eclipse/uprotocol/uri/serializer/LongUriSerializer.java

```diff
@@ -234,6 +234,10 @@ private static UResource parseFromString(String resourceString) {
         if (resourceMessage != null) {
             uResourceBuilder.setMessage(resourceMessage);
         }
+
+        if (resourceName.contains("rpc") && resourceInstance != null && resourceInstance.contains("response")) {
```

Comment:
IMHO these resourceName.contains("rpc") && resourceInstance != null && resourceInstance.contains("response")
should be extracted to their own methods since they are useful

Reply:
Yes it is in UriValidator.isRpcResponse() so I can just call that actually.

Reply:
Will keep as is, cannot use the validator (as we are building the object now)